drivers/ethernet/eth_stm32_hal_v2.c

```diff
@@ -580,6 +594,36 @@ int eth_stm32_hal_init(const struct device *dev)
 #endif
 	heth->Init.RxBuffLen = ETH_STM32_RX_BUF_SIZE;
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32mp13_ethernet)
+	/* Map memory region for DMA descriptor and buffer as non cacheable */
+	k_mem_map_phys_bare(&desc_uncached_addr,
+			    DT_REG_ADDR(ETH_DMA_REGION),
+			    DT_REG_SIZE(ETH_DMA_REGION),
+			    K_MEM_PERM_RW | K_MEM_DIRECT_MAP | K_MEM_ARM_NORMAL_NC);
+
+	heth->Init.ClockSelection = cfg->clockselection;
+#endif
+
+#if DT_INST_CLOCKS_HAS_NAME(0, eth_ker)
+	/* Turn on DCMIPP peripheral clock */
+	ret = clock_control_configure(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+				      (clock_control_subsys_t)&cfg->pclken_ker,
+				      NULL);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure ETH kernel clock. Error %d", ret);
+		return ret;
+	}
+#endif
+
+#if DT_INST_CLOCKS_HAS_NAME(0, mac_clk)
+	ret = clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+			       (clock_control_subsys_t)&cfg->pclken_mac);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure mac clock. Error %d", ret);
+		return ret;
+	}
+#endif
```

Comment:
why don't we move it to the enableing of the other clocks in `eth_initialize`? We already set the ptp clock there too, which also is v2 exclusive